src/components/issue-state-badge.component.js

```diff
@@ -1,52 +1,26 @@
 import React from 'react';
-import { Text, View, StyleSheet } from 'react-native';
 
-import { colors, fonts, normalize } from 'config';
+import { StateBadge } from 'components';
 
 type Props = {
   issue: Object,
   isMerged: boolean,
 };
 
-const styles = StyleSheet.create({
-  badge: {
-    padding: 12,
-    paddingTop: 3,
-    paddingBottom: 3,
-    borderRadius: 20,
-  },
-  mergedIssue: {
-    backgroundColor: colors.purple,
-  },
-  openIssue: {
-    backgroundColor: colors.green,
-  },
-  closedIssue: {
-    backgroundColor: colors.red,
-  },
-  text: {
-    fontSize: normalize(12),
-    ...fonts.fontPrimarySemiBold,
-    color: colors.white,
-  },
-});
-
 export const IssueStateBadge = ({ issue, isMerged }: Props) => {
-  return (
-    <View
-      style={[
-        styles.badge,
-        isMerged && styles.mergedIssue,
-        issue.state === 'open' && !isMerged && styles.openIssue,
-        issue.state === 'closed' && !isMerged && styles.closedIssue,
-      ]}
-    >
-      {isMerged && <Text style={styles.text}>Merged</Text>}
+  let type = '';
+  let title = '';
+
+  if (isMerged) {
+    type = 'merged';
+    title = 'Merged';
+  } else if (issue.state === 'open') {
+    type = 'open';
+    title = 'Open';
+  } else if (issue.state === 'closed') {
+    type = 'closed';
+    title = 'Closed';
+  }
 
-      {!isMerged &&
-        <Text style={styles.text}>
-          {issue.state === 'open' ? 'Open' : 'Closed'}
-        </Text>}
-    </View>
-  );
+  return <StateBadge type={type} text={title} />;
```

Comment:
Do we need this additional component at all? Looks like `IssueStateBadge` takes in `issue` and `isMerged` and  just calls `StateBadge`. I think we should be okay with just having one of these components (`StateBadge` for example that takes in `isMerged` and `issue` as props to define `type` and `title`. 

I guess you wanted to see the logic abstracted for `Merged`, `Open` etc.. badges for issues instead of regular badges. I think a nicer alternative would be to let `StateBadge` to take in a prop called `issue` OR take two props: `type` and `title`. 

Let me know what you think!